lib/ansible/modules/cloud/amazon/ec2_vpc_net.py

```diff
@@ -396,6 +427,8 @@ def main():
             for cidr in to_add:
                 changed = True
                 connection.associate_vpc_cidr_block(CidrBlock=cidr, VpcId=vpc_id)
+        if ipv6_cidr:
```

Comment:
There can only be one IPv6 CIDR block per VPC, so I think in addition to this there should also be a check that there isn't already an Ipv6CidrBlockAssociationSet for the VPC. If you run your example task twice you'll get a traceback rather than changed: False the second time, and this would fix that. Do you mind also adding a test to cover this?